light-node/src/commands/start.rs

```diff
@@ -108,53 +88,103 @@ impl config::Override<LightNodeConfig> for StartCmd {
         &self,
         mut config: LightNodeConfig,
     ) -> Result<LightNodeConfig, FrameworkError> {
-        if !self.rpc_addr.is_empty() {
-            config.rpc_address = self.rpc_addr.to_owned();
+        // Todo figure out if other options would be reasonable to overwrite via CLI arguments.
+        if let Some(addr) = self.listen_addr {
+            config.rpc_config.listen_addr = addr;
         }
-
         Ok(config)
     }
 }
-
-/*
- * The following is initialization logic that should have a
- * function in the lite crate like:
- * `subjective_init(height, vals_hash, store, requester) -> Result<(), Error`
- * it would fetch the initial header/vals from the requester and populate a
- * trusted state and store it in the store ...
- * TODO: this should take traits ... but how to deal with the State ?
- * TODO: better name ?
- */
-async fn subjective_init(
-    height: Height,
-    vals_hash: Hash,
-    store: &mut MemStore,
-    req: &RPCRequester,
-) -> Result<(), Error> {
-    if store.get(height).is_ok() {
-        // we already have this !
-        return Ok(());
+impl StartCmd {
+    fn assert_init_was_run() {
+        // TODO: handle errors properly:
+        let primary_db_path = app_config().light_clients.first().unwrap().db_path.clone();
+        let db = sled::open(primary_db_path).unwrap_or_else(|e| {
+            println!("[error] could not open database: {}", e);
```

Comment:
Don't we have proper logging facilities through abcissa?

Reply:
Good point. I think it does.

Reply:
![image](https://user-images.githubusercontent.com/7441766/87223623-d0869d00-c37e-11ea-9365-667c71ae1a3e.png)